jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/ExplorerToolWindow.kt

```diff
@@ -29,27 +30,32 @@ import software.aws.toolkits.jetbrains.core.credentials.ProjectAccountSettingsMa
 import software.aws.toolkits.jetbrains.core.credentials.ProjectAccountSettingsManager.AccountSettingsChangedNotifier.AccountSettingsEvent
 import software.aws.toolkits.jetbrains.core.explorer.ExplorerDataKeys.SELECTED_RESOURCE_NODES
 import software.aws.toolkits.jetbrains.core.explorer.ExplorerDataKeys.SELECTED_SERVICE_NODE
+import software.aws.toolkits.jetbrains.core.explorer.actions.CopyArn
 import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerNode
 import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerResourceNode
 import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerServiceRootNode
 import software.aws.toolkits.jetbrains.services.lambda.LambdaFunctionNode
 import software.aws.toolkits.jetbrains.services.lambda.execution.remote.RemoteLambdaLocation
+import software.aws.toolkits.jetbrains.ui.tree.AsyncTreeModel
+import software.aws.toolkits.jetbrains.ui.tree.StructureTreeModel
 import software.aws.toolkits.resources.message
 import java.awt.Component
-import java.awt.event.MouseAdapter
 import java.awt.event.MouseEvent
 import javax.swing.JPanel
 import javax.swing.JTree
 import javax.swing.tree.DefaultMutableTreeNode
-import javax.swing.tree.DefaultTreeModel
+import javax.swing.tree.TreeModel
 
 class ExplorerToolWindow(private val project: Project) : SimpleToolWindowPanel(true, true), AccountSettingsChangedNotifier {
     private val actionManager = ActionManagerEx.getInstanceEx()
     private val projectAccountSettingsManager = ProjectAccountSettingsManager.getInstance(project)
 
     private val treePanelWrapper: Wrapper = Wrapper()
     private val errorPanel: JPanel
-    private var awsTree: Tree? = null
+    private val awsTreeModel = AwsExplorerTreeStructure(project)
+    private val structureTreeModel = StructureTreeModel(awsTreeModel, project)
+    private var awsTree = createTree(AsyncTreeModel(structureTreeModel, true, project))
```

Comment:
Thought we didn't have to replace the tree, why does this have to be a `var`?

Reply:
It can be a val now